rust/noosphere/src/ffi/headers.rs

```diff
@@ -3,8 +3,18 @@ use safer_ffi::prelude::*;
 #[derive_ReprC(rename = "ns_headers")]
 #[repr(opaque)]
 /// @class ns_headers_t
+/// An opaque struct representing name/value headers.
 ///
-/// TBD
+/// Headers are used in ns_sphere_file_t to assign metadata, like content type.
```

Comment:
An example of "class" examples. As we only have one big example for now, unclear if having individual (untested??) examples is better. Can we run non-rust tests as a part of rustdocs? Hmm

Reply:
I would actually prefer not to have code examples in docs for the very reason that we can't easily keep them up to date. Better to have actual tests!

Reply:
agreed, examples running against CI way preferable at this point, will remove this